Delete URLUtilsReadOnly docs, redirect to WorkerLocation docs

sideshowbarker opened the pull request:
https://github.com/whatwg/html/commit/32a7a20 completely removed the `URLUtilsReadOnly` interface and moved/inlined/hoisted its members up into the `WorkerLocation` interface. So this change deletes all the `URLUtilsReadOnly` docs, and redirects them to the `WorkerLocation` docs.

Related BCD change: https://github.com/mdn/browser-compat-data/pull/10347

wbamberg commented on files/en-us/web/api/url_api/index.html:
```diff
@@ -74,7 +74,7 @@ <h2 id="URL_API_interfaces">URL API interfaces</h2>
 </div>
 
-<p>Older versions of the specification included an interface called {{domxref("URLUtilsReadOnly")}}, which has since been merged into the {{domxref("WorkerLocation")}} interface.</p>
+<p>Older versions of the specification included an interface called <code>URLUtilsReadOnly</code>, which has since been merged into the {{domxref("WorkerLocation")}} interface.</p>
```
Does this sentence still make sense? I'm not really sure what it's adding here.

sideshowbarker replied:
> Does this sentence still make sense? I'm not really sure what it's adding here.

You’re right, it adds nothing at this point — so I’ve pushed an amended commit that includes dropping that completely.

wbamberg reviewed (commented):
This looks fine, although I had a question about one sentence that didn't seem very useful any more.

But it looks like all the redirects work, and the redirect entries match the removed files.

sideshowbarker committed:
Delete URLUtilsReadOnly docs, redirect to WorkerLocation docs

https://github.com/whatwg/html/commit/32a7a20 completely removed the
URLUtilsReadOnly interface and moved/inlined/hoisted its members up
into the WorkerLocation interface. So this change deletes all the
URLUtilsReadOnly docs, and redirects them to the WorkerLocation docs.

Related BCD change: https://github.com/mdn/browser-compat-data/pull/10347

chrisdavidmills reviewed (approved):
Looks like @sideshowbarker 's latest changes satisfied @wbamberg 's query, so I think I'll go ahead and merge this. Thanks Mike!
